src/main/java/graphql/schema/DataFetchingEnvironmentImpl.java

```diff
@@ -66,7 +69,7 @@ private DataFetchingEnvironmentImpl(Builder builder) {
         this.locale = builder.locale;
         this.operationDefinition = builder.operationDefinition;
         this.document = builder.document;
-        this.variables = builder.variables == null ? Collections.emptyMap() : builder.variables;
+        this.variables = builder.variables == null ? ImmutableMapWithNullValues.empty() : builder.variables;
```

Comment:
Maybe renamed `ImmutableMapWithNullValues.empty()` =to `ImmutableMapWithNullValues.emptyMap()` - this would be clearer in my book and also better on static import

```
this.variables = builder.variables == null ? emptyMap() : builder.variables;
```